frontend/test/test.helpers.ts

```diff
@@ -35,41 +40,54 @@ export const spyOnAxiosApiCallWithFile = (api, method) => {
 };
 
 export const createTestStore = () => {
-	let waitingActions: Action[] | string[] = [];
+	let waitingActions: WaitAction[] = [];
 	let resolvePromise;
+	let debugActions = false;
 
 	const sagaMiddleware = createSagaMiddleware();
 	const middlewares = applyMiddleware(sagaMiddleware);
 
-	const discountMatchingActions =  (action: Action) => {
+	const discountMatchingActions =  (state: object, action: Action, previousState: object) => {
 		const waitingAction = waitingActions[0];
 	
-		if (action.type === waitingAction || isEqual(waitingAction, action)) {
+		const successStep = isFunction(waitingAction) ? waitingAction(state, action, previousState) : false;
+
+		if (action.type === waitingAction || isEqual(waitingAction, action) || successStep) {
 			waitingActions.shift();
 		}
 	
 		return waitingActions.length === 0;
 	};
 
-	const store = createStore(combineReducers(
-		{
-			...reducers,
-			spyActions: (state, action) => {
-
-				if (discountMatchingActions(action) && resolvePromise) {
-					resolvePromise(true);
-					resolvePromise = null;
-				}
-				return {};
-			}
+	const mainReducer = combineReducers(reducers);
+
+
+	const store = createStore((state: any, action) =>{
+		const st = mainReducer(state, action);		
+
+		if (debugActions) {
+			console.log(JSON.stringify({
+				dispatchedAction: action,
+				waitingAction: waitingActions[0],
+			}, null, '\t'));
+		}
+
+		if (discountMatchingActions(st, action, state) && resolvePromise) {
+			resolvePromise(true);
+			resolvePromise = null;
 		}
-	), middlewares);
+		
+		return st;
+	}		
+	, middlewares);
 	
-	const waitForActions = (func, waitActions) =>  { 
+	const waitForActions:WaitForActions = (dispatchingfunction, waitActions, debug?: boolean) =>  { 
```

Comment:
Is the idea that if we want to debug a test that uses `waitForActions` we append `true` to it?

Reply:
Is just to make it do a console log. After doing that I added a way to run it through vscode so maybe this is no longer necessary.